### PR TITLE
Add Certificate Type

### PR DESCRIPTION
**Pull request by sumandari**
Please see #1371

- [x] Part 1 : Create a new model for certification course types and an admin page for creating / deleting types.
- [x] Part 2 : In the certification menu, add a place for certification managers to manage the list of certificate types. Only certification managers can see this menu.
- [x] Part 3 : Setting the attendance type on the course.
- [x] Part 4 : Update the template to get the certification type text from the course model.

### CODE REVIEW

**Comment by sumandari** [comment]
Demo:

### Only super users and certification managers can access certification types menu

https://user-images.githubusercontent.com/40058076/147897606-ca3c16c9-8660-4a8b-aba0-4152bcb4a04f.mp4

---

### Manage certificate types to be enabled on the projects
https://user-images.githubusercontent.com/40058076/147897607-8dbe7753-0035-4523-b8ed-05b626008cc9.mp4

---

File: django_project/certification/views/certificate_type.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            objects.filter(project=project).values_list(
+            'certificate_type', flat=True)
+        return context
+
+    def get_queryset(self):
+        """Return certificate_types for a project."""
+
+        project = get_object_or_404(Project, slug=self.kwargs['project_slug'])
+        qs = ProjectCertificateType.objects.filter(project=project)
+        return qs
```
Comment (dimasciput) [inline django_project/certification/views/certificate_type.py]:
```suggestion
        return ProjectCertificateType.objects.filter(project=project)
```

File: django_project/certification/views/certificate.py [hunk 2]
```diff
@@ -948,8 +956,16 @@ def preview_certificate(request, **kwargs):
 
         current_site = request.META['HTTP_HOST']
 
-        generate_pdf(
-            response, project, course, attendee, certificate, current_site)
+        if certificate_type_id:
+            certificate_type = CertificateType.objects.get(
```
Comment (dimasciput) [inline django_project/certification/views/certificate.py]: @sumandari please add a check here

File: django_project/certification/models/certificate_type.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    )
+
+    order = models.IntegerField(
+        blank=True,
+        null=True,
+        unique=True
+    )
+
+    def __str__(self):
+        return self.name
```
Comment (sumandari) [inline django_project/certification/models/certificate_type.py:42]: check codecov:
This one is covered in test_CRUD_CertificateType
https://github.com/kartoza/prj.app/pull/1383/files#diff-dd4bdce4503bf9fafe9947650eb5d398f8c1c897642b377dec9ef0847eb8c8e0R145

File: django_project/certification/views/certificate.py (hunk 2 above)
Comment (sumandari) [reply django_project/certification/views/certificate.py]: I have updated it with try catch block for this get query

**Comment by dimasciput** [comment]
LGTM please merge @sumandari

**Pull request merged**